scripts/postinstall.js

```diff
@@ -0,0 +1,9 @@
+'use strict'
+
+/* eslint-disable no-console */
+
+const execSync = require('child_process').execSync
+
+if (__dirname.indexOf('/node_modules/') !== -1) {
```

Comment:
Wouldn't a better check be whether or not the `.git` directory is present?

Reply:
What are the advantages?

Reply:
All developer machines are going to have a `.git` directory and no machines installing it from `npm` will, wheras the `node_modules` directory could vary depending on such weird things as `npm link` and people working inside a `node_modules` directory in order to make it easier to `require` things.

Reply:
A lot of developers will be installing it from `npm`, so this is not exactly true. I feel like both approaches have roughly the same limitations.

Reply:
How about using one of the `npm_*` environment variables? I think at the end of the day depending on the file system will always end up causing issues.

Reply:
That makes the intent much clearer too, so I'm quite cool with that.

Reply:
Looks like npm is a bit too clever and gives each package its own information, so you still don't know if you are the root package or a dependency. Back to the drawing board.

Reply:
Switched to comparing the following 2 environment variables:

* **INIT_CWD**: Folder where `npm install` was run.
* **PWD**: From any of a dependency files, the root folder of the dependency.

If they match it means `npm install` was run directly from the library, not as a dependency of another library.